examples/python/pytorch/resnet152_DDP_training.py

```diff
@@ -8,49 +8,52 @@
 import torch.multiprocessing as mp
 import torchvision.transforms as transforms
 import torchvision.models as models
+
 import time
 import pandas as pd
 
 
 def setup(rank, world_size):
-    os.environ['MASTER_ADDR'] = 'localhost'
-    os.environ['MASTER_PORT'] = '12355'
+    print("Running init_process_group...")
     dist.init_process_group("nccl", rank=rank, world_size=world_size)
+    print("Finished init_process_group...")
 
 def cleanup():
     dist.destroy_process_group()
 
-def train(local_rank, args):
-    rank = args.nr * args.gpus + local_rank	
+def train(gpu, args):
+    rank = args.nr * args.gpus + gpu	
     setup(rank, args.world_size)
     transform = transforms.Compose([
                 torchvision.transforms.Resize(224),
                 transforms.RandomHorizontalFlip(),
                 transforms.ToTensor(),
                 transforms.Normalize(mean=[0.485, 0.456, 0.406],std=[0.229, 0.224, 0.225])
             ])
-    batch_size = 77
+    batch_size = args.batchsize
     train_dataset = torchvision.datasets.CIFAR10('./datasets/',transform=transform,download=True)
     sampler = torch.utils.data.distributed.DistributedSampler(train_dataset,num_replicas=args.world_size,rank=rank)
     trainloader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, num_workers=2,sampler=sampler)
-    model = resnet152()
-    torch.cuda.set_device(local_rank)
-    model.cuda()
 
-    model = nn.parallel.DistributedDataParallel(model,device_ids=[local_rank])
-    print("Setting optimizer")
+    model = models.resnet152()
+    torch.cuda.set_device(gpu)
+    model.cuda()
+    print("GPU initialization")
+    dummy_input = torch.randn(1, 3,224,224, dtype=torch.float).to(gpu)
```

Comment:
Why is this needed?

Reply:
It is mentioned here that the performance of GPUs is different when they just started from when are already running. The dummy input there is what is done on this article:
https://deci.ai/the-correct-way-to-measure-inference-time-of-deep-neural-networks/